contracts/base/ExecutionHelper.sol

```diff
@@ -53,6 +53,21 @@ contract ExecutionHelper is IExecutionHelperEventsAndErrors {
         }
     }
 
+    /// @notice Executes a call to a target address with specified value and data.
+    /// Same as _execute but without return data for gas optimization.
+    function _executeNoReturndata(address target, uint256 value, bytes calldata callData) internal virtual {
+        assembly {
+            let result := mload(0x40)
+            calldatacopy(result, callData.offset, callData.length)
+            if iszero(call(gas(), target, value, result, callData.length, 0, 0)) {
+                // Bubble up the revert if the call reverts.
+                returndatacopy(result, 0x00, returndatasize())
+                revert(result, returndatasize())
+            }
+            mstore(0x40, add(result, callData.length)) //allocate memory
```

Comment:
didn't get this part

Reply:
since we used memory storage, need to manually refresh free data pointer

Reply:
size of the allocated memory will be callData.length?